app/models/history_entry.rb

```diff
@@ -77,7 +77,7 @@ def self.element_ids_with(options = {})
   
     # use first key of options as attribute
     ref_attribute = options.keys.first
-    ref_date      = options[:at].to_time
+    ref_date      = options[:at].to_time.end_of_day
```

Comment:
Do not use `to_time` on Date objects, because they know nothing about the time zone in use.